src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,252 @@
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Input,
+  OnDestroy,
+  Output,
+  ViewChild,
+  AfterViewInit,
+  ChangeDetectionStrategy,
+} from '@angular/core';
+import { T } from '../../t.const';
+import { FormControl } from '@angular/forms';
+import { BehaviorSubject, combineLatest, from, Observable } from 'rxjs';
+import {
+  tap,
+  map,
+  debounceTime,
+  switchMap,
+  withLatestFrom,
+  startWith,
+} from 'rxjs/operators';
+import { TaskService } from '../tasks/task.service';
+import { Router } from '@angular/router';
+import { TODAY_TAG } from '../tag/tag.const';
+import { Project } from '../project/project.model';
+import { Tag } from '../tag/tag.model';
+import { ProjectService } from '../project/project.service';
+import { TagService } from '../tag/tag.service';
+import { Task } from '../tasks/task.model';
+import { blendInOutAnimation } from 'src/app/ui/animations/blend-in-out.ani';
+import { AnimationEvent } from '@angular/animations';
+import { SearchItem, SearchQueryParams } from './search-bar.model';
+import { getWorklogStr } from '../../util/get-work-log-str';
+import { devError } from 'src/app/util/dev-error';
+
+@Component({
+  selector: 'search-bar',
+  templateUrl: './search-bar.component.html',
+  styleUrls: ['./search-bar.component.scss'],
+  animations: [blendInOutAnimation],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class SearchBarComponent implements AfterViewInit, OnDestroy {
+  @Input() isElevated: boolean = false;
+  @Output() blurred: EventEmitter<any> = new EventEmitter();
+
+  @ViewChild('inputEl') inputEl!: ElementRef;
+  @ViewChild('searchForm') searchForm!: ElementRef;
+
+  T: typeof T = T;
+  isLoading$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(true);
+  taskSuggestionsCtrl: FormControl = new FormControl();
+  filteredIssueSuggestions$: Observable<SearchItem[]> = new Observable();
+  isArchivedTasks: boolean = false;
+  isArchivedTasks$: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  tooManyResults: boolean = false;
+
+  private _attachKeyDownHandlerTimeout?: number;
+  private _tasks: Task[] = [];
+  private _projects: Project[] = [];
+  private _tags: Tag[] = [];
+  private searchableItems$: Observable<SearchItem[]> = this.isArchivedTasks$.pipe(
+    switchMap((isArchivedTasks) => this._loadTasks$(isArchivedTasks)),
+    withLatestFrom(this._projectService.list$, this._tagService.tags$),
+    tap(([tasks, projects, tags]) => this._keepForLater(tasks, projects, tags)),
```

Comment:
It's cleaner to do this in a new subscriptiion rather than as a side effect here. If you have multiple subscriptions to searchableItems$ this get's executed for every single one even though it is just needed once.